chain/network/src/peer_manager.rs

```diff
@@ -357,6 +358,23 @@ impl PeerManagerActor {
             warn!(target: "network", "Unknown account {} in peers, not supported indirect routing", account_id);
         }
     }
+
+    fn process_announce_account(&mut self, peer_id: PeerId, account_id: AccountId, signature: Signature) -> AnnounceAccountResult {
+        if self.account_peers.contains_key(&account_id) {
+            // This announcement was already received.
+            return AnnounceAccountResult::IgnoreAnnounce;
+        }
+
+        // Verify the signature.
+        if !verify(account_id.as_bytes(), &signature, &peer_id.pub_key()) {
+            return AnnounceAccountResult::InvalidSignature;
+        }
+
+        // TODO: Establish direct connection with peer if not in `self.active_peers`
+        self.account_peers.insert(account_id, peer_id);
+
+        return AnnounceAccountResult::BroadcastAccount;
```

Comment:
when we broadcast account, if we want to become a relay of it, we should report our "distance" to the account.
not yet sure if we want to - but we can sign this with our `peer_id` pretty much collecting this signatures as proof of relaying.

Reply:
@ilblackdragon I've been putting some thought on this, and I think this will be not enough in the long term. If some nodes get disconnected (intentionally or not) the network might split (adversaries can make some elaborated plan ddos-ing some nodes).

Though your proposotion is way simpler and effective (in good scenarios) a better choice is use kademlia.